compiler/src/errors.rs

```diff
@@ -18,17 +20,19 @@ pub enum CompilerErrorKind {
   Unassignable,
   UnknownOperator,
 
+  TypeMismatch { type1: Type, type2: Type },
+
   Unimplemented,
 }
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct CompilerError {
   pub kind: CompilerErrorKind,
-  pub index: usize,
+  pub range: Range,
 }
 
 impl CompilerError {
-  pub fn new(kind: CompilerErrorKind, index: usize) -> Self {
-    Self { kind, index }
+  pub fn new(kind: CompilerErrorKind, range: Range) -> Self {
+    Self { kind, range }
   }
 }
```

Comment:
Would it work to keep `LexerError`, `ParserError` and `CompilerError` separate, and instead try to standardize error handling across the crates?
```rust
struct DiagnosticsHandler<E, W> {
  pub errors: Vec<E>,
  pub warnings: Vec<W>,
}
```
We probably want to keep the error handling process of each stage separate, since we should halt if there are errors in the current stage before moving on to the next one.

Reply:
Resolved by 536acff5dde53dfc5fe80af859009e84b9ab5156